### ISSUE TITLE
Wrap options interferes badly with escodegen.sourceMapWithCode

### ISSUE THREAD

**Issue by kirill-konshin** [gfranko/amdclean#71]
As a follow up for https://github.com/gfranko/amdclean/issues/65

If `escodegen` has `sourceMapWithCode:true` then any given `wrap` other than `false` will produce ugly results with `[object Object]` instead of the code.

The cause is the following code block which does not respect Escodegen's output nuance: when it has `sourceMapWithCode:true` its output is an object `{code: ..., map: ...}`.

``` js
// If there is a wrap option specified
if (_.isObject(options.wrap)) {
  if (_.isString(options.wrap.start) && options.wrap.start.length) {
    generatedCode = options.wrap.start + generatedCode;
  }
  if (_.isString(options.wrap.end) && options.wrap.end.length) {
    generatedCode = generatedCode + options.wrap.end;
  }
}
```

Basically, if amdclean works with enabled source maps, this wrapping functionality is meaningless because it corrupts the source map (which will be unaware of changes to the code). Appropriate way of sourcemap-respectful wrapping is something like https://github.com/RReverser/gulp-wrap-js. In general, amdclean should not wrap anything at all, it is not this plugin's responsibility.

**Comment by kirill-konshin** [comment]
I recommend to take a look on https://github.com/estools/estemplate which is used by `gulp-wrap-js`. This can solve issues with wrapping.

Or drop the wrapping functionality completely.
